src/Controls/src/Core/Handlers/Items/iOS/CarouselViewController.cs

```diff
@@ -70,6 +81,23 @@ public override void ViewDidLoad()
 			base.ViewDidLoad();
 		}
 
+		internal override void AttachingToWindow()
+		{
+			base.AttachingToWindow();
+			
+			if(_initialOrientation == null)
+			{
+				_initialOrientation = DeviceDisplay.MainDisplayInfo.Orientation;
+				DeviceDisplay.MainDisplayInfoChanged += OnDisplayInfoChanged;
+			}
+		}
+
+		internal override void DettachingFromWindow()
+		{
+			base.DettachingFromWindow();
+			TearDown();
```

Comment:
I don't think we can just outright call `TearDown` here

Or we need to move some of the cod that's in the ctor into `AttachingToWindow`

for example this is unsubscrived from

```C#
carousel.Scrolled -= CarouselViewScrolled;
```

Inside of TearDown, but subscribed to in the ctor. 

Just because something is detaching from the window, doesn't mean it won't get re-attached. 

For example, when a modal is pushed on top of the current screen. The VC below that modal will get unattached.

Reply:
Yeah , you  right, got eager that actually never gets called, let me move the subscribe to the correct place to.